aws_lambda_builders/workflows/nodejs_npm_esbuild/actions.py

```diff
@@ -74,10 +75,15 @@ def execute(self) -> None:
         :raises lambda_builders.actions.ActionFailedError: when esbuild packaging fails
         """
         esbuild_command = EsbuildCommandBuilder(
-            self._scratch_dir, self._artifacts_dir, self._bundler_config, self._osutils, self._manifest
+            self._working_directory, self._output_directory, self._bundler_config, self._osutils, self._manifest
         )
 
         if self._should_bundle_deps_externally():
+            check_minimum_esbuild_version(
+                minimum_version_required=MINIMUM_VERSION_FOR_EXTERNAL,
+                working_directory=self._working_directory,
+                subprocess_esbuild=self._subprocess_esbuild,
+            )
```

Comment:
This used to be a separate action. Since we need to check the version every time that we want to mark things as `--external` then I think we should do the check when/before marking things as external (here), instead of having a separate action that would always have to be appended before this one. This cleans things up a bit, but it's also safer as we don't have to remember that we need a check action before this bundle action.